sigmf/sigmffile.py

```diff
@@ -511,13 +533,23 @@ def validate(self):
         version = self.get_global_field(self.VERSION_KEY)
         validate.validate(self._metadata, self.get_schema())
 
-    def archive(self, name=None, fileobj=None):
+    def archive(self, sigmffile_name=None, archive_name=None, fileobj=None):
```

Comment:
I'm not sure about this, but what would you think about removing the `sigmffile_name` argument here, and requiring the `name` argument for the creation of a SigMFFile? Imagine the situation:
- SigMFFile is created without a `name` argument, so `self.name == None`
- SigMFFile.archive is called without a `sigmffile_name` argument, so SigMFArchive gets called with `name = None`
- SigMFArchive expects each input SigMFFile object have a `self.name` set

I'm not sure what happens in this case, but I think it would be reasonable to require the name to be set _somewhere_, or implement a clearly-documented behavior for generating unique names if one is not provided.

Reply:
For this situation, a SigMFFileError will be raised. It checks for a SigMFFile name in archive.py. 

`sigmf.error.SigMFFileError: the `name` attribute must be set to pass to `SigMFArchive``

Do you think that is sufficient or should name be required in SigMFFile?

Reply:
added test to ensure exception raised in 89242c8

Reply:
If the name is not required, this approach should be used in `tofile` as well:

The optional `name` parameter you added to SigMFFile indicates that it will be used only when passing to SigMFArchive. Let the `name` instance variable be used inside both `tofile()` and `archive()`, which would both require a file path to be provided as a parameter. That way, the `name` parameter can stay optional for `SigMFFile`, UNTIL you try to save it somewhere, at which point it must be set. This could get confusing if the user provides a direct path to the intended location of the saved file, rather than a path to the directory in which to save it.

Making the `name` required, and using it in both `tofile()` and `archive()` seems like a much more clear solution to me. I assume that most SigMFFile objects are created with the intent of eventually saving them with a name, so I think this is reasonable.

Reply:
Made name required in SigMFFIle constructor in 0c503ab